addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4417,22 +4417,17 @@ var SnippetsMenu = Widget.extend({
      * Preview on mobile.
      */
     _onMobilePreviewClick() {
-        this.trigger_up('request_mobile_preview');
-
-        // TODO refactor things to make this more understandable -> on mobile
-        // edition, update the UI. But to do it properly and inside the mutex
-        // this simulates what happens when a snippet option is used.
+        this._execWithLoadingEffect(async () => this.trigger_up('request_mobile_preview'), false);
```

Comment:
Instead of doing two `_execWithLoadingEffect` in a row, why not put the trigger_up inside the existing one? If this is because the trigger_up triggers another call to `_execWithLoadingEffect`, then this cannot work either way because the second `_execWithLoadingEffect` would be done second or third depending on the current queue :grimacing: To be investigated.

Reply:
`request_mobile_preview` actually calls a handler from an OWL component (wysiwyg_adapter), which toggles on & off the `websiteService.context.isMobile`. So no other `_execWithLoadingEffect` involved.

> why not put the trigger_up inside the existing one?

This seems to work indeed (I thought I had already tested it, but maybe it was a slight variation which doesn't work):
```js
        this._execWithLoadingEffect(async () => {
            this.trigger_up("request_mobile_preview");
            new Promise(resolve => setTimeout(() => {
                this.trigger_up('snippet_option_update', {
                    onSuccess: () => resolve(),
                });
            }));
        }, false);
```